Add functionality to render a show All button

himanshm opened the pull request:
Render a definite number of collections initially with a show All button 
Render all collections as soon as the button is clicked and hide the button when all collections are rendered.
On branch content-library-modal
Changes to be committed:
	modified:   .eslintrc.json
	new file:   listeners/actions/fetch-all-collections-action.js
	modified:   listeners/actions/index.js
	modified:   services/strapi/index.js
	modified:   user-interface/activity-collection/index.js
	modified:   user-interface/app-home/unauthorised-user-view.js
	modified:   user-interface/modals/content-library-modal.js
	
Appropriate comments have been provided at required places.

Commit:
Add functionality to render a showAll button

Render a definite number of collections initially with a showAll button
Render all collections as soon as the button is clicked and hide the
button when all collections are rendered.
On branch content-library-modal
Changes to be committed:
	modified:   .eslintrc.json
	new file:   listeners/actions/fetch-all-collections-action.js
	modified:   listeners/actions/index.js
	modified:   services/strapi/index.js
	modified:   user-interface/activity-collection/index.js
	modified:   user-interface/app-home/unauthorised-user-view.js
	modified:   user-interface/modals/content-library-modal.js

disha1202 commented on .eslintrc.json:
```diff
@@ -12,6 +12,7 @@
     "no-console": "off",
-    "eqeqeq": "off"
+    "eqeqeq": "off",
+    "operator-linebreak": "off"
```
Remove operator line break

himanshm replied:
Done!

disha1202 commented on services/strapi/index.js:
```diff
@@ -54,9 +54,13 @@ async function getSurveyQuestions() {
+async function getActivityCollections(params) {
   try {
-    const url = `${baseUrl}/activity-collections?populate[thumbnail][populate]=*&populate[activities][populate]=*`;
+    // If no endpoint is provided(endpoint is empty), The initial collection is rendered through
```
Update the comment

himanshm replied:
Done!

disha1202 commented on services/strapi/index.js:
```diff
@@ -69,9 +73,28 @@ async function getActivityCollections() {
+
+  try {
+    const url =
+      `${baseUrl}/activity-collections/${collectionId}?${collectionQueryParam}`;
```
Move in a single line

disha1202 commented on services/strapi/index.js:
```diff
@@ -54,9 +54,13 @@ async function getSurveyQuestions() {
+    // If no endpoint is provided(endpoint is empty), The initial collection is rendered through
+    // intialEndpoint.
+    const initialEndpoint = '/activity-collections?populate=*&pagination[start]=0&pagination[limit]=3';
+    const showAllCollectionEndpoint = 'activity-collections?populate[thumbnail][populate]=*';
+    const url = params?.showAll ? `${baseUrl}/${showAllCollectionEndpoint}` : `${baseUrl}${initialEndpoint}`;
```
```suggestion
    const showAllCollectionEndpoint = '/activity-collections?populate[thumbnail][populate]=*';
    const url = `${baseUrl}/${params?.showAll ? showAllCollectionEndpoint : initialEndpoint}`;
```

himanshm replied:
Done!

Commit:
Remove line breaks
Remove line-break rule from eslintrc
update the url variable
On branch content-library-modal
Changes to be committed:
	modified:   .eslintrc.json
	modified:   services/strapi/index.js

disha1202 commented on user-interface/modals/content-library-modal.js:
```diff
@@ -20,16 +19,9 @@ async function getContentLibraryModal(body) {
-        thumbnail_url: `${baseUrl}${selectedCollection.attributes.thumbnail.data.attributes.url}`,
+        type: 'image',
+        image_url: `${baseUrl}${selectedCollection.attributes.thumbnail.data.attributes.url}`,
+        alt_text: 'inspiration',
```
Should we make this alt text dynamic?

Commit:
Change image alt text
On branch content-library-modal
Changes to be committed:
	modified:   user-interface/modals/content-library-modal.js
